packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -168,3 +171,131 @@ export namespace ProblemDecorator {
         (left.data.severity || Number.MAX_SAFE_INTEGER) - (right.data.severity || Number.MAX_SAFE_INTEGER);
 
 }
+
+@injectable()
+export class ProblemTabBarDecorator implements TabBarDecorator {
+
+    readonly id = 'theia-problem-tabbar-decorator';
+
+    protected emitter: Emitter<(titles: Title<Widget>[]) => void>;
+
+    @inject(ProblemManager)
+    protected readonly problemManager: ProblemManager;
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    @postConstruct()
+    protected init(): void {
+        this.emitter = new Emitter();
+        this.problemManager.onDidChangeMarkers(() => this.fireDidChangeDecorations((titles: Title<Widget>[]) => this.collectDecorators(titles)));
```

Comment:
Why do you need to call the same function when fire an event and in `decorate`? Can event just be of `void` type and let `tabbardecorator` to call `decorate` again?